### PR TITLE
feat: Extend progress stepper results

### PR DESCRIPTION
**Pull request by Venefilyn**
This changes the existing work on `ResultProgressStep` to include
status handling for cancelled and failed. Further improving UX by
ensuring that the status reporting is consistent across the page

Fixes

RELEASE NOTES BEGIN
Improve UX for task progression results 
RELEASE NOTES END

### CODE REVIEW

File: frontend/src/app/Results/ResultsPageCoprDetails.tsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ export const ResultsPageCoprDetails: React.FC<ResultsPageCoprDetailsProps> = ({
             submittedTime={data.build_submitted_time}
             startTime={data.build_start_time}
             finishedTime={data.build_finished_time}
+            status={data.status === "failure" ? "fail" : "success"}
```
Comment (lbarcziova) [inline frontend/src/app/Results/ResultsPageCoprDetails.tsx:64]: with these we get success for errors, right? Could the condition for them be added as well, as they are also quite often I would say.

**Review by lbarcziova — COMMENTED** [review]
nice improvement!

File: frontend/src/app/Results/ResultProgressStep.tsx [hunk 2]
```diff
@@ -2,15 +2,58 @@
...
+  Tooltip,
+  ProgressStepProps,
+} from "@patternfly/react-core";
 import { Timestamp } from "../utils/Timestamp";
 import { prettyFormat } from "@imranbarbhuiya/duration";
 
+/**
+ * Different statuses that we should map within the component
+ */
+type AcceptedStatuses = "success" | "fail" | "cancelled";
```
Comment (lachmanfrantisek) [inline frontend/src/app/Results/ResultProgressStep.tsx:17]: There is also skipped:

image

https://packit-dashboard-pr-415.surge.sh/results/propose-downstream/8709

**Review by lachmanfrantisek — COMMENTED** [review]
Nice! Thanks a lot!

**Comment by Venefilyn** [comment]
@lachmanfrantisek @lbarcziova Added a few more statuses and fixed so we have skipped, unknown, and cancelled mapped

File: frontend/src/app/Results/ResultsPageSyncReleaseRuns.tsx [hunk 3]
```diff
@@ -242,6 +242,27 @@ const ResultsPageSyncReleaseRuns: React.FC<ResultsPageSyncReleaseRunsProps> = ({
...
+   *
+   * TODO (@Venefilyn): change the statuses to match API
+   *
+   * @param {string} status - list of statuses from sync release
+   * @return {*}  {AcceptedStatuses}
+   */
+  function getSyncReleaseStatus(status: string): AcceptedStatuses {
+    switch (status) {
+      case "error":
+        return "fail";
```
Comment (lbarcziova) [inline frontend/src/app/Results/ResultsPageSyncReleaseRuns.tsx:256]: I think this should be added also to the other mapping functions

**Review by lbarcziova — APPROVED** [review]
otherwise LGTM :+1:

File: frontend/src/app/Results/ResultsPageSyncReleaseRuns.tsx (hunk 3 above)
Comment (Venefilyn) [reply frontend/src/app/Results/ResultsPageSyncReleaseRuns.tsx:256]: What do you mean by to the other mapping functions? A case for error?

Comment (lbarcziova) [reply frontend/src/app/Results/ResultsPageSyncReleaseRuns.tsx:256]: the `getSRPMStatus`, `getCoprBuildStatus` etc.

Comment (Venefilyn) [reply frontend/src/app/Results/ResultsPageSyncReleaseRuns.tsx:256]: Ahh gotcha, didn't see that before. Just added it

image

**Pull request merged**